File: 2025-05_Spring-101-Java-Maven/src/main/java/com/happn/worshop/application/UserController.java
package com.happn.worshop.application;

import com.happn.worshop.domain.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.UUID;

import static com.happn.worshop.application.UserDtoMapper.toDto;

@RestController
@RequiredArgsConstructor
@RequestMapping("/users")
public class UserController {

    private final UserService userService;

    @GetMapping("/{id}")
    public UserDto findById(@PathVariable UUID id) {

        return userService.findById(id)
                .map(UserDtoMapper::toDto)
                .orElseThrow();
    }

    @PostMapping
    public UserDto create(@RequestBody UserCreateDto userCreateDto) {
        return toDto(userService.createUser(userCreateDto.name(), userCreateDto.email()));
    }
}

File: 2025-05-06_Spring-101-Kotlin-Gradle/src/main/kotlin/com/happn/workshop/application/UserController.kt
package com.happn.workshop.application

import com.happn.workshop.domain.UserService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.util.*

@RestController
@RequestMapping("/users")
class UserController(private val userService: UserService) {

    @GetMapping("/{id}")
    fun findById(@PathVariable id: String): UserDto =
        userService.findById(UUID.fromString(id))?.toDto() ?: throw IllegalStateException("User not found")

    @PostMapping
    fun save(@RequestBody userCreation: UserCreateDto): String = userService.save(user = userCreation.toDomain())

}